datafusion/common/src/utils.rs

```diff
@@ -492,6 +496,44 @@ pub fn list_ndims(data_type: &DataType) -> u64 {
     }
 }
 
+/// Create an new empty array based on the given data type.
```

Comment:
Technically this list isn't empty, is it ? It contains a single `Null` value

Also, it seems like this handles all data types, not just a list. I wonder if the same thing could be done by making a null scalar like

```rust
let null_scalar = ScalarValue::try_from(&data_type)?;
null_scalar.to_array_of_size(1)
```
🤔